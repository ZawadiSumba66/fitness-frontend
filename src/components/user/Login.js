import { connect } from 'react-redux';
import { useState } from 'react';
import PropTypes from 'prop-types';
import { Link } from '@reach/router';
import { loginUser } from '../../actions/user_action';
import store from '../../store';
import Flash from './Flash';

const Login = ({ errors }) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const handleSubmit = (e) => {
    e.preventDefault();
    const user = {
      email,
      password,
    };
    store.dispatch(loginUser(user));
  };

  return (

    <div className="d-flex flex-column signup-form justify-content-center mx-5">
      <div>
        <i className="fas fa-heartbeat" />
        <span className="font-weight-bold">KEEP IT FIT</span>
      </div>
      <p> Welcome Back! Lets continue with our journey of keeping fit</p>
      <Flash />
      <div className="errors">
        {errors ? (
          <div>
            <li className="text-danger">Invalid email or password</li>
          </div>
        ) : (
          ''
        )}
      </div>
      <form onSubmit={handleSubmit}>
        <input
          value={email}
          onChange={(e) => setEmail(e.target.value)}
          type="email"
          name="email"
          placeholder="Enter your email"
          className="form-control mt-3"
        />
        <input
          value={password}
          onChange={(e) => setPassword(e.target.value)}
          type="password"
          name="password"
          placeholder="Password"
          className="form-control mt-3"
        />
        <input
          type="submit"
          value="Log in"
          className="button-orange btn text-light text-uppercase font-weight-bold w-100 mt-4"
        />
      </form>
      <p className="pt-5 text-center">Don`t have an account?</p>
      <Link
        to="/signup"
        className="text-center"
      >
        SignUp
      </Link>
    </div>
  );
};

const mapStateToProps = (state) => ({
  errors: state.userReducer.login_backend_error,
});

Login.propTypes = {
  errors: PropTypes.string.isRequired,
};

export default connect(mapStateToProps)(Login);
