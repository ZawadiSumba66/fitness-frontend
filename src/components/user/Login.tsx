import { connect } from 'react-redux';
import { useState } from 'react';
import { Link } from '@reach/router';
import { loginUser } from '../../actions/user_action';
import store from '../../store';
import Flash from './Flash';

type LoginProps = {
  errors: string;
};

const Login = ({ errors }: LoginProps) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    const user = {
      email,
      password,
    };
    store.dispatch(loginUser(user));
  };

  return (

    <div className="d-flex flex-column signup-form justify-content-center px-10 md:px-60 fitness-image">
      <div>
        <i className="fas fa-heartbeat" />
        <span className="font-bold">KEEP IT FIT</span>
      </div>
      <p className="text-white font-bold"> Welcome Back! Lets continue with our journey of keeping fit</p>
      <Flash />
      <div className="errors">
        {errors ? (
          <div>
            <li className="text-danger">Invalid email or password</li>
          </div>
        ) : (
          ''
        )}
      </div>
      <form onSubmit={handleSubmit}>
        <input
          value={email}
          onChange={(e) => setEmail(e.target.value)}
          type="email"
          name="email"
          placeholder="Enter your email"
          className="form-control mt-3"
        />
        <input
          value={password}
          onChange={(e) => setPassword(e.target.value)}
          type="password"
          name="password"
          placeholder="Password"
          className="form-control mt-3"
        />
        <button
          type="submit"
          className="button-orange btn text-light text-uppercase font-weight-bold w-100 mt-4"
          value="Sign up"
        >
          login
        </button>
      </form>
      <p className="pt-5 text-center text-white">Don`t have an account?</p>
      <Link
        to="/signup"
        className="text-center text-dark font-bold"
      >
        SignUp
      </Link>
    </div>
  );
};

type LoginState = {
  userReducer: {
    login_backend_error: string,
  }
};

const mapStateToProps = (state: LoginState) => ({
  errors: state.userReducer.login_backend_error,
});

export default connect(mapStateToProps)(Login);
