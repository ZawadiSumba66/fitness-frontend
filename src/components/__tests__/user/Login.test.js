import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../../store';
import Login from '../../user/Login';

it('matches the login page snapshot', () => {
  const tree = renderer.create(<Provider store={store}><Login /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
