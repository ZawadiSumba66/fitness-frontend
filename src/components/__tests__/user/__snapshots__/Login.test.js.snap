// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the login page snapshot 1`] = `
<div
  className="d-flex flex-column signup-form justify-content-center px-10 md:px-60 fitness-image"
>
  <div>
    <i
      className="fas fa-heartbeat"
    />
    <span
      className="font-bold"
    >
      KEEP IT FIT
    </span>
  </div>
  <p
    className="text-white font-bold"
  >
     Welcome Back! Lets continue with our journey of keeping fit
  </p>
  <div />
  <div
    className="errors"
  >
    
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="form-control mt-3"
      name="email"
      onChange={[Function]}
      placeholder="Enter your email"
      type="email"
      value=""
    />
    <input
      className="form-control mt-3"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <button
      className="button-orange btn text-light text-uppercase font-weight-bold w-100 mt-4"
      type="submit"
      value="Sign up"
    >
      Login
    </button>
  </form>
  <p
    className="pt-5 text-center text-white"
  >
    Don\`t have an account?
  </p>
  <a
    className="text-center text-dark font-bold"
    href="/signup"
    onClick={[Function]}
  >
    SignUp
  </a>
</div>
`;
